repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala

```diff
@@ -52,8 +52,30 @@ object Preprocessor{
     val TypeDef = DefProc("type"){ case m: G#TypeDef => m.name }
 
     val PatVarDef = Processor { case (name, code, t: G#ValDef) =>
+      //Function to lift expressions into anonymous functions so they will be JITed
+      def wrap(code: String)={
+        import fastparse._
+        import scalaparse.Scala._
+        val par = P( ( `implicit`.? ~ `lazy`.? ~ ( `var` | `val` ) ~! BindPattern.rep1("," ~! Pass) ~ (`:` ~! Type).?).! ~ (`=` ~! StatCtx.Expr.!) )
+        val Result.Success((lhs, rhs), _) =
+          par.parse(code).asInstanceOf[Result.Success[(String, String)]]
```

Comment:
No need `asInstanceOf` here: the pattern match should already fail if its a failure